lib/block_view.go

```diff
@@ -4726,135 +5590,780 @@ func (bav *UtxoView) _connectUpdateProfile(
+
+		// TODO: This field is deprecated and should be deleted.
+		newProfileEntry.StakeMultipleBasisPoints = txMeta.NewStakeMultipleBasisPoints
+
+		// The StakeEntry is always left unmodified here.
+
+	} else {
+		// When there's no pre-existing profile entry we need to do more
+		// checks.
+		if len(txMeta.NewUsername) == 0 {
+			return 0, 0, nil, RuleErrorProfileUsernameTooShort
+		}
+		// We allow users to create profiles without a description or picture
+		// in the consensus code. If desired, frontends can filter out profiles
+		// that don't have these fields.
+		//
+		// Creator percentage and stake multiple are sufficiently checked above.
+
+		// In this case we need to set all the fields using what was passed
+		// into the transaction.
+
+		// If below block height, user transaction public key.
+		// If above block height, use ProfilePublicKey if available.
+		profileEntryPublicKey := txn.PublicKey
+		if blockHeight > ParamUpdaterProfileUpdateFixBlockHeight {
+			profileEntryPublicKey = profilePublicKey
+		}
+
+		newProfileEntry = ProfileEntry{
+			PublicKey:   profileEntryPublicKey,
+			Username:    txMeta.NewUsername,
+			Description: txMeta.NewDescription,
+			ProfilePic:  txMeta.NewProfilePic,
+
+			CoinEntry: CoinEntry{
+				CreatorBasisPoints: txMeta.NewCreatorBasisPoints,
+
+				// The other coin fields are automatically set to zero, which is an
+				// appropriate default value for all of them.
+			},
+
+			// TODO(DELETEME): This field is deprecated and should be deleted because we're
+			// not allowing staking to profiles.
+			StakeMultipleBasisPoints: txMeta.NewStakeMultipleBasisPoints,
+			// TODO(DELETEME): This field is deprecated and should be deleted because we're
+			// not allowing staking to profiles.
+			StakeEntry: NewStakeEntry(),
+		}
+	}
+	// At this point the newProfileEntry should be set to what we actually
+	// want to store in the db.
+
+	if verifySignatures {
+		// _connectBasicTransfer has already checked that the transaction is
+		// signed by the top-level public key, which we take to be the poster's
+		// public key.
+	}
+
+	// Delete the old profile mappings. Not doing this could cause a username
+	// change to have outdated mappings, among other things.
+	if prevProfileEntry != nil {
+		bav._deleteProfileEntryMappings(prevProfileEntry)
+	}
+
+	// Save the profile entry now that we've updated it or created it from scratch.
+	bav._setProfileEntryMappings(&newProfileEntry)
+
+	// Add an operation to the list at the end indicating we've updated a profile.
+	utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+		Type:             OperationTypeUpdateProfile,
+		PrevProfileEntry: prevProfileEntry,
+	})
+
+	return totalInput, totalOutput, utxoOpsForTxn, nil
+}
+
+func (bav *UtxoView) _connectCreateNFT(
+	txn *MsgBitCloutTxn, txHash *BlockHash, blockHeight uint32, verifySignatures bool) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+
```

Comment:
I would error if bav.GlobalParamsEntry.MaxCopiesPerNFT == 0 to feature flag this.